peps/pep-0747.rst

```diff
@@ -200,14 +200,20 @@ that represent a type assignable to ``str | None``::
   err1: TypeForm[str | None] = str | int  # Error
   err2: TypeForm[str | None] = list[str | None]  # Error
 
-By this same definition, ``TypeForm[Any]`` describes a type form object
-that represents the type ``Any`` or any type that is assignable to ``Any``.
-Since all types in the Python type system are assignable to ``Any``,
-``TypeForm[Any]`` describes the set of all type form objects
-evaluated from all valid type expressions.
+By this same definition, ``TypeForm[object]`` describes a type form object that
+represents the type ``object`` or any type that is assignable to ``object``.
+Since all types in the Python type system are assignable to ``object``,
+``TypeForm[object]`` describes the set of all type form objects evaluated from
+all valid type expressions.
 
-The type expression ``TypeForm``, with no type argument provided, is
-equivalent to ``TypeForm[Any]``.
+The type expression ``TypeForm``, with no type argument provided, signals that
+the expression assigned to this annotated type should be interpreted as a type
+expression, resulting in a ``TypeForm`` type, but the type parameter of that
+``TypeForm`` type should be inferred from the type expression. For example,
+these two assignments are equivalent::
+
+  tf1: TypeForm = int | None
+  tf2: TypeForm[int | None] = int | None
```

Comment:
This is similar to how bare `ClassVar` and bare `Final` are handled, and to me it seems this behavior is already implied by various examples in the PEP itself. It seems more useful than having bare `TypeForm` simply mean `TypeForm[Any]`.

Reply:
On second thought, this behavior doesn't work for `TypeForm` since it is an actual type, not a type qualifier. A function parameter annotated with bare `TypeForm` needs to have some interpretation as a type, without the benefit of inference from any particular callsite. I'll change this back.

(I would rather have `TypeForm` mean `TypeForm[object]` rather than `TypeForm[Any]`, but I guess we have the precedent of other generic types defaulting to `Any`, and we should probably stick with it.)

Reply:
Yeah, I'm glad you came to that realization and conclusion. I was going to make the same argument when I read your comment here.